pkg/typesystem/typesystem.go

```diff
@@ -126,23 +126,45 @@ func Difference(base *openfgapb.Userset, sub *openfgapb.Userset) *openfgapb.User
 }
 
 type TypeSystem struct {
+	// [objectType] => typeDefinition
 	typeDefinitions map[string]*openfgapb.TypeDefinition
-	modelID         string
-	schemaVersion   string
+	// [objectType] => [relationName] => relation
+	relations     map[string]map[string]*openfgapb.Relation
+	modelID       string
+	schemaVersion string
 }
 
 // New creates a *TypeSystem from an *openfgapb.AuthorizationModel.
 // It assumes that the input model is valid. If you need to run validations, use NewAndValidate.
 func New(model *openfgapb.AuthorizationModel) *TypeSystem {
 	tds := make(map[string]*openfgapb.TypeDefinition, len(model.GetTypeDefinitions()))
+	relations := make(map[string]map[string]*openfgapb.Relation, len(model.GetTypeDefinitions()))
+
 	for _, td := range model.GetTypeDefinitions() {
 		tds[td.GetType()] = td
+		tdRelations := make(map[string]*openfgapb.Relation, len(td.GetRelations()))
+
+		for relation, rewrite := range td.GetRelations() {
+			r := &openfgapb.Relation{
+				Name:     relation,
+				Rewrite:  rewrite,
+				TypeInfo: &openfgapb.RelationTypeInfo{},
+			}
+
+			if metadata, ok := td.GetMetadata().GetRelations()[relation]; ok {
+				r.TypeInfo.DirectlyRelatedUserTypes = metadata.GetDirectlyRelatedUserTypes()
+			}
```

Comment:
Will it ever by !ok?  Should we throw panic for these cases?

Reply:
if it's a 1.0 model, line 155 should be ignored